### DISCUSSION TITLE
Allow OTP expiry interval to be set

### DISCUSSION THREAD

**Discussion by zehawki** [django-otp/django-otp#130] [Q&A]
How is the OTP expiry interval to be changed? It seems to have some undocumented default at the moment, maybe 30 days?

**Comment by psagers** [comment]
Can you be more specific? Expiration can mean different things (or nothing at all) for different devices. TOTPDevice generates a new code every [30 seconds](https://django-otp-official.readthedocs.io/en/stable/overview.html#django_otp.plugins.otp_totp.models.TOTPDevice.step) by default. EmailDevice generates tokens that are valid for [300 seconds](https://django-otp-official.readthedocs.io/en/stable/extend.html#django_otp.models.SideChannelDevice.generate_token) by default.

**Comment by zehawki** [comment]
I get logged out of the Django admin periodically (after some days). I was thinking its due to this package, but I guess I'm mistaken. Its probably due to the session middleware.
